Lib/test/datetimetester.py

```diff
@@ -62,6 +62,19 @@ def test_constants(self):
         self.assertEqual(datetime.MINYEAR, 1)
         self.assertEqual(datetime.MAXYEAR, 9999)
 
+    def test_all(self):
+        """Test that __all__ only points to valid attributes."""
```

Comment:
Hmm, I think there's a simpler way to test this, just do:

```python
from datetime import *
```

This will throw an AttributeError if you have an invalid attribute defined there.

Example (`test.py`):
```python
__all__ = ['a']
b = 1
```
```python
>>> from test import *
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'test' has no attribute 'a'
```

cc @pganssle?

Reply:
@ammaraskar `from x import *` is only valid at the top level of a file, so this cannot be executed in a test.

Reply:
@pganssle Ah, is this already the most viable way to test the validity of the attributes then?

Reply:
Well, it's the best one I could think of. There may be better tests for `__all__`, but given that this module already does some funky and apparently fragile stuff with the import mechanism, I think it's probably best not to try to get too fancy - particularly since it's not *that* important of a test.